### PR TITLE
fix(aws-ec2): add dependency on gateway attachment for public routes

### LINKED ISSUE #1140: PublicRoute does not depend on gateway attachment
Got the following error when deploying ec2.VpcNetwork: `route table rtb-xxxx and network gateway igw-yyyy belong to different networks`.  This appears to be a race condition when the gateway attachment has not yet completed by the time the public route is created.  The PublicRoute needs to depend on the gateway attachment:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-dependson.html#gatewayattachment

Example synthesized route:
```
  VPCPublicSubnet2DefaultRouteB7481BBA:
    Type: 'AWS::EC2::Route'
    Properties:
      RouteTableId:
        Ref: VPCPublicSubnet2RouteTable6F1A15F1
      DestinationCidrBlock: 0.0.0.0/0
      GatewayId:
        Ref: VPCIGWB7E252D3
```
Route element needs a "DependsOn" for the gateway attachment

### PR DESCRIPTION
**Pull request by clareliguori**
Fixes #1140.  Currently there is a race condition when creating the public subnet routes for the ec2.VpcNetwork construct.  CloudFormation can attempt to create the public subnet routes to the IGW before the IGW is attached to the VPC.  This change adds a dependency to the public routes on the IGW attachment,

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

BREAKING CHANGE: Method signature of VpcPublicSubnet.addDefaultIGWRouteEntry changed in order to add a dependency on gateway attachment completing before creating the public route to the gateway. Instead of passing a gateway ID string, pass in a cloudformation.InternetGatewayResource object and a cloudformation.VPCGatewayAttachmentResource object.

### CODE REVIEW

File: packages/@aws-cdk/aws-ec2/lib/vpc.ts [hunk 1]
```diff
@@ -520,12 +520,15 @@ export class VpcSubnet extends VpcSubnetRef implements cdk.ITaggable {
     });
   }
 
-  protected addDefaultRouteToIGW(gatewayId: string) {
-    new cloudformation.RouteResource(this, `DefaultRoute`, {
+  protected addDefaultRouteToIGW(
```
Comment (eladb) [inline packages/@aws-cdk/aws-ec2/lib/vpc.ts]: Could you please add some jsdocs here and also mention that a dependency will be created?

Comment (clareliguori) [reply packages/@aws-cdk/aws-ec2/lib/vpc.ts]: Done

**Comment by eladb** [comment]
Please switch the order of paragraphs in the commit description so that the BREAKING CHANGE paragraph is last (as per [conventionalcommits](https://www.conventionalcommits.org/en/v1.0.0-beta.2/). 

The reason we are particular about this is because we generate our changelogs from these, and if there’s a breaking change, everything below it will be taken.

**Comment by clareliguori** [comment]
Done

**Pull request merged**